### DISCUSSION TITLE
Process decorators without transpiling them

### DISCUSSION THREAD

**Discussion by simonihmig** [babel/babel#13927] [Q&A]
I would like to process modules with babel that use (legacy) decorators, but without transpiling those. So the build output should contain the transformations I explicitly added, but the decorator invocations should be left untouched. But this doesn't seem to work.

Without any decorator-related plugins added, Babel (expectedly) fails to parse the modules:

```
[!] (plugin babel) SyntaxError: [...]: Support for the experimental syntax 'decorators-legacy' isn't currently enabled (7:3):
```

When I add `@babel/plugin-syntax-decorators`, I would expect this to be solved, i.e. that the plugin enables Babel to parse the file correctly, but won't transform the decorator invocations, as the docs say...

> Syntax only
> It's unlikely you want to use this plugin directly as it only enables Babel to parse this syntax. Instead, use plugin-proposal-decorators to both parse and transform this syntax.

However that doesn't work either, with the plugin added I get this:

```
[!] Error: Unexpected character '@'
```

Only adding `@babel/plugin-proposal-decorators` enables Babel to process the file, but then the decorators are transformed, which I wanted to avoid.

Unfortunately I cannot provide a reproduction in the REPL, as you cannot provide the necessary plugin configuration there.

I am not sure if this is supposed to work as I would want it. If so, I can file this as a real bug report, but I first wanted to check whether my expectations here are correct.

**Comment by nicolo-ribaudo** [comment]
You can try using `@babel/plugin-syntax-decorators`. Every `-proposal-` plugin has the corresponding `-syntax-` plugin that only enables parsing.

**Comment by simonihmig** [comment]
@nicolo-ribaudo ok, but that's what I tried as explained above, and it failed...

> When I add @babel/plugin-syntax-decorators [...]

> However that doesn't work either, with the plugin added I get this:
```
[!] Error: Unexpected character '@'
```
